tests/test_request_fingerprinter.py

```diff
@@ -0,0 +1,371 @@
+            "https://example.com",
+            {"httpRequestMethod": "POST"},
+            b"\xe1\xf3&2R%\x0c\x82mf\x88E\x11L\x05w+\xa6V\xcb",
+        ),
+        (
+            "https://example.com",
+            {"httpResponseBody": True},
+            b"e\x1e\xd3J0ya_\xca\xc3\xa0\xbe'h\x0ff*\xa6b\xf2",
+        ),
+        (
+            "https://example.com",
+            {"httpResponseHeaders": True},
+            b"\xcc^\x0e$\xa7\xe5\x97\xb8\xbf\x7f0\xa3\xec\xf5B\\\xe1h\x1c\xee",
+        ),
+        (
+            "https://example.com",
+            {"browserHtml": True},
+            b"\xb2\x8e\x98\xa9\xa2\xf2\xa6\x96\x01\xf6\x1dYa\xf7\xdf\xc2\xe5>x\x11",
+        ),
+        (
+            "https://example.com",
+            {"screenshot": True},
+            b"\x8a\xd1\x1fut\x99\xf1\xc4\xcc\xa8\xfd\xd9\x7f\x1fY\xf8\xdf/'\xb3",
+        ),
+        (
+            "https://example.com",
+            {"screenshotOptions": {"format": "png"}},
+            b"\xe2\xba\xeb\x16\xb9\xd4\x117\x19\xac\x7f\xb3\x17\xf5\xf6\xfc\x9e\x94l\xcf",
+        ),
+        (
+            "https://example.com",
+            {"geolocation": "US"},
+            b"#\xe2\\\xce\xb88\xf8\xb4\x19\xa09KL\xe4\x87\x80\x00\x00A7",
+        ),
+        (
+            "https://example.com",
+            {"javascript": False},
+            b"\x1c!\x89\xfc\xadd\xb3\xbf-_\x97\xca\xc0g\xbdo\xee\xdc\xdfo",
+        ),
+        (
+            "https://example.com",
+            {"actions": [{"action": "click", "selector": ".button"}]},
+            b"\x83\xfa\x04\xfal\xc6d(\xe1\x06\xf1>b\xed\xbe\xb1\xf2\xac5E",
+        ),
+    ),
+)
+def test_known_fingerprints(url, params, fingerprint):
+    """Test that known fingerprints remain the same, i.e. make sure that we do
+    not accidentally modify fingerprints with future implementation changes."""
+    crawler = get_crawler()
+    fingerprinter = create_instance(
+        ScrapyZyteAPIRequestFingerprinter, settings=crawler.settings, crawler=crawler
+    )
+    request = Request(url, meta={"zyte_api": params})
+    actual_fingerprint = fingerprinter.fingerprint(request)
+    assert actual_fingerprint == fingerprint
+
+
+def test_metadata():
+    settings = {"JOB": "1/2/3"}
+    crawler = get_crawler(settings_dict=settings)
+    job_fingerprinter = create_instance(
+        ScrapyZyteAPIRequestFingerprinter, settings=crawler.settings, crawler=crawler
+    )
+
+    crawler = get_crawler()
+    no_job_fingerprinter = create_instance(
+        ScrapyZyteAPIRequestFingerprinter, settings=crawler.settings, crawler=crawler
+    )
+
+    request1 = Request("https://example.com", meta={"zyte_api": {"echoData": "foo"}})
+    request2 = Request("https://example.com", meta={"zyte_api": True})
+
+    fingerprint1 = job_fingerprinter.fingerprint(request1)
+    fingerprint2 = no_job_fingerprinter.fingerprint(request2)
+    fingerprint3 = job_fingerprinter.fingerprint(request1)
+    fingerprint4 = no_job_fingerprinter.fingerprint(request2)
```

Comment:
Could we combine `fingerprint1` and `fingerprint3` together? as well as `fingerprint2` and `fingerprint4`? as they are essentially the same.

Or was the intent to have a permutation like this?
```python
    fingerprint1 = job_fingerprinter.fingerprint(request1)
    fingerprint2 = no_job_fingerprinter.fingerprint(request2)
    fingerprint3 = job_fingerprinter.fingerprint(request2)
    fingerprint4 = no_job_fingerprinter.fingerprint(request1)
```